app/[locale]/[...uriSegments]/page.tsx

```diff
@@ -0,0 +1,176 @@
+import StaffPageTemplate from "@/components/templates/StaffPage";
+import EventPageTemplate from "@/components/templates/EventPage";
+
+const pickFeaturedImage = async (
+  locale: string,
+  image?: any,
+  cantoAsset?: any,
+  pressReleaseId?: string
+): Promise<OpenGraph["images"] | undefined> => {
+  if (pressReleaseId) {
+    const releaseOpenGraphImage = await getReleaseOpenGraph(
+      locale,
+      pressReleaseId
+    );
+
+    return releaseOpenGraphImage;
+  }
+
+  if (cantoAsset) {
+    const {
+      width,
+      height,
+      url: { directUrlPreview },
+    } = cantoAsset;
+    const url = resizeCantoImage(directUrlPreview, 800);
+
+    return { url, width, height };
+  }
+
+  if (image) {
+    const { url, width, height, altText: alt } = image;
+
+    return { url, width, height, alt };
+  }
+};
+
+export async function generateMetadata(
+  { params: { locale, uriSegments }, searchParams = {} }: UriSegmentProps,
+  parent: ResolvingMetadata
+): Promise<Metadata> {
+  const uri = uriSegments.join("/");
+  let previewToken: string | undefined;
+
+  if (draftMode().isEnabled) {
+    previewToken = Array.isArray(searchParams.preview)
+      ? searchParams.preview[0]
+      : searchParams?.preview;
+  }
+
+  const { entry } = await getEntryMetadataByUri(uri, locale, previewToken);
+
+  if (!entry) {
+    notFound();
+  }
+
+  const {
+    title,
+    description,
+    image = [],
+    cantoAssetSingle = [],
+    pressReleaseId,
+  } = entry;
+
+  const featuredImage = await pickFeaturedImage(
+    locale,
+    image[0],
+    cantoAssetSingle[0],
+    pressReleaseId
+  );
+  const previousImages = (await parent).openGraph?.images || [];
+
+  return {
+    title,
+    description: striptags(description),
+    openGraph: {
+      images: featuredImage || previousImages,
+    },
+  };
+}
+
+export async function generateStaticParams({
+  params: { locale },
+}: LocaleProps) {
+  const data = await getEntriesByLocale(locale);
+
+  return data.entries.map(({ uri }) => {
+    return {
+      locale,
+      uriSegments: uri.split("/"),
+    };
+  });
+}
+
+const sectionMap = {
+  events: EventPageTemplate,
+  // galleryItems: GalleryPageTemplate,
+  glossaryTerms: GlossaryPageTemplate,
+  news: NewsPageTemplate,
+  searchResults: SearchPageTemplate,
+  slideshows: SlideshowPageTemplate,
+  staffProfiles: StaffPageTemplate,
+  // userProfilePage: UserProfilePageTemplate,
+};
+
+const UriSegmentsPage: FunctionComponent<UriSegmentProps> = async ({
+  params: { locale, uriSegments },
+  searchParams,
+}) => {
+  const uri = uriSegments.join("/");
+  let previewToken: string | undefined;
+
+  if (draftMode().isEnabled) {
+    previewToken = Array.isArray(searchParams.preview)
+      ? searchParams.preview[0]
+      : searchParams?.preview;
+  }
+
+  const entrySectionType = await getEntrySectionByUri(
+    uri,
+    locale,
+    previewToken
+  );
+
+  // Handle 404 if there is no data
+  if (!entrySectionType) {
+    notFound();
```

Comment:
the previous static file purging has been removed. App directory uses cached data instead of static built pages so there is no pre-built content to purge in a 404 scenario